shared/linkify.ts

```diff
@@ -11,6 +11,8 @@ export type LinkPart = {
 	link: string;
 };
 
+const webplus_scheme_chars = /^[a-z]+:/i;
```

Comment:
this is an invalid regex... Will match web+SpOnGeBoB: which is not valid

> The custom scheme has only lowercase [ASCII](https://developer.mozilla.org/en-US/docs/Glossary/ASCII) letters in its name.

https://developer.mozilla.org/en-US/docs/Web/API/Navigator/registerProtocolHandler#permitted_schemes

Reply:
that documentation is for the call to registerProtocolHandler, not for the `href`s/links themselves (which get normalized by the browser before being passed onto the OS/the handler).

Reply:
well, https://html.spec.whatwg.org/multipage/iana.html#web+-scheme-prefix agrees, so while my link was poorly chosen, the statement is still true I think

Reply:
we can refer to RFC 3986, section 3.1: https://www.rfc-editor.org/rfc/rfc3986#section-3.1

> Scheme names consist of a sequence of characters beginning with a letter and followed by any combination of letters, digits, plus ("+"), period ("."), or hyphen ("-").  Although schemes are case- insensitive, the canonical form is lowercase and **documents that specify schemes must do so with lowercase letters**.

(emphasis added)

there is also:

> An implementation should accept uppercase letters as equivalent to lowercase in scheme names (e.g., allow "HTTP" as well as "http") for the sake of robustness but should only produce lowercase scheme names for consistency.

and in practice this is what browsers do, for `a href` they normalize the scheme name before processing it further.